pkg/broker/util.go

```diff
@@ -250,23 +267,26 @@ func parametersToSchema(plan apb.Plan) Schema {
 		},
 	}
 
-	return s
+	return s, nil
 }
 
-func extractProperties(params []apb.ParameterDescriptor) map[string]*schema.Schema {
+func extractProperties(params []apb.ParameterDescriptor) (map[string]*schema.Schema, error) {
 	properties := make(map[string]*schema.Schema)
 
 	var patternRegex *regexp.Regexp
-	var err error
-
 	for _, pd := range params {
 		k := pd.Name
 
+		t, err := getType(pd.Type)
```

Comment:
Why not just check to see if the type is UnspecifiedType? I'm not sure I like the error on the `getType`. Is it really an error on getType? It looked for one, didn't fine a good one, said it was unspecified.

```
t := getType(pd.Type)
if t == []schema.PrimitiveType{schema.UnspecifiedType} {
    return properties
}
```

Reply:
I think it is reasonable that an ununderstandable type is an error. Especially since we can not use that type for it's intended purpose. If you feel strongly though, I can change the initial error here instead of in `getType` function.